### DISCUSSION TITLE
`backup` followed by `forget --prune` in a schedule

### DISCUSSION THREAD

**Discussion by fdw** [creativeprojects/resticprofile#226] [Q&A] [answered]
I would like to start `backup` followed by `forget --prune` in a schedule. I can start both with a schedule each, but then it depends on luck which one starts first, and the other runs into a lock.
Additionally, `retention` seems to be deprecated.

What is the correct way to start first and and directly the other in one schedule?

**Comment by creativeprojects** [comment, marked as answer]
Sorry if it's not being very clear:

The `retention` section is **not** deprecated.

What is deprecated is scheduling the `retention` on its own.

So, the way it works is:
- you can schedule a `backup` followed by a `retention` (in the same schedule)
- you can schedule a `forget`. Typically a scheduled `forget` can have different options than the `retention` section.

**Reply by jkellerer** [reply]
One addition: ATM, you need to add `after-backup=true` in the retention section to let it run after backup. If you want to enable prune, add `prune=true`.

@ @creativeprojects : think we might need a documentation page on retention. This part isn't so trivial.

**Reply by creativeprojects** [reply]
I'll add that to my TODO list 😉

**Reply by fdw** [reply]
Thank you very much 🙂

Just to make sure I understand this correctly: I schedule the `backup`, and by adding `after-backup=true` to the `retention` section, it will automatically also be scheduled after each backup?

Also, is there an easy way to use the same configuration as `forget`, or do I have to duplicate it?

**Reply by jkellerer** [reply]
Yes correct, `after-backup=true` in `retention` will also run `restic forget` with the configuration done inside the `retention` section, whenever you run `resticprofile backup` (also when it is scheduled).

If you'd like to run `resticprofile forget`, you configure the `forget` section. This is then running `restic forget` using the configuration done in the `forget` section.

If for some reason you need the same configuration in both section (you probably never would but just in case), you can use [templates](https://creativeprojects.github.io/resticprofile/configuration/templates/index.html) (or [mixins](https://creativeprojects.github.io/resticprofile/configuration/inheritance/index.html#mixins)) to define it once and reuse it.

**Reply by fdw** [reply]
Awesome, thank you 🙂

May I ask one more question, out of curiosity? I can trigger `check` from `backup` with `check-after` and `check-before`. Why have you decided to use a new section `retention` instead of `forget-after` and `forget-before` in `backup`? From what I can see so far is that the only difference is that you had two configurations of `forget`, one for stand-alone calls and one for `retention`.

However, with `forget-after`, it would be easier to see that another action is triggered after the backup, and it's consistent with `check-after`. You could even use the `forget` commands and web-hooks, which `retention` doesn't have (unfortunately).

Please don't take this the wrong way, I expect that you thought long about my suggestion and decided that a `retention` configuration is better 🙂 I only want to understand your decision, not question it.

**Reply by jkellerer** [reply]
We have an outstanding feature idea to introduce aliases that can span over multiple commands and that are meant to replace “-before” and “-after” (auto aliases would remain backward compatibility). So yes there is awareness 😀

However retention and forget will exist in parallel, since sometimes you want to be able to remove individual snapshots. You can do this with `resticprofile myprofile.forget snapshotid` (basically you don’t want the normal retention logic when tweaking the history manually)

**Reply by fdw** [reply]
Thanks 🙂 No more questions 😉
